tests/d3d12.c

```diff
@@ -35726,6 +35726,152 @@ static void test_vertex_shader_stream_output(bool use_dxil)
     destroy_test_context(&context);
 }
 
+static void test_index_buffer_edge_case_stream_output(void)
+{
+    D3D12_GRAPHICS_PIPELINE_STATE_DESC pso_desc;
+    ID3D12Resource *counter_buffer, *so_buffer;
+    ID3D12GraphicsCommandList *command_list;
+    D3D12_STREAM_OUTPUT_BUFFER_VIEW sobv;
+    ID3D12Resource *upload_buffer;
+    struct test_context_desc desc;
+    ID3D12Resource *index_buffer;
+    struct resource_readback rb;
+    D3D12_INDEX_BUFFER_VIEW ibv;
+    struct test_context context;
+    ID3D12CommandQueue *queue;
+    unsigned int counter, i;
+    const struct vec4 *data;
+    ID3D12Device *device;
+    HRESULT hr;
+
+    static const D3D12_SO_DECLARATION_ENTRY so_declaration[] =
+    {
+        {0, "SV_Position", 0, 0, 4, 0},
+    };
+    static const struct vec4 expected_output[] =
+    {
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+        { 3.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f,-3.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f}, /* For the case where we are rendering with NULL index. The first vertex is always picked. */
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f}, /* For DrawInstanced with NULL index buffer, which should work just fine. */
+        { 3.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f,-3.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f}, /* For the case where we are rendering with NULL GPU VA for index. */
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f,-3.0f, 0.0f, 1.0f}, /* For the case where we are rendering with UNKNOWN index format. It is actually R16_UINT. */
+        { 3.0f, 1.0f, 0.0f, 1.0f},
+        {-1.0f, 1.0f, 0.0f, 1.0f},
+    };
+    unsigned int strides[] = {16};
+    static const uint16_t index_data[3] = { 2, 1, 0 };
+
+    memset(&desc, 0, sizeof(desc));
+    desc.root_signature_flags = D3D12_ROOT_SIGNATURE_FLAG_ALLOW_STREAM_OUTPUT;
+    desc.no_pipeline = true;
+    if (!init_test_context(&context, &desc))
+        return;
+
+    device = context.device;
+    command_list = context.list;
+    queue = context.queue;
+
+    index_buffer = create_upload_buffer(device, sizeof(index_data), index_data);
+
+    init_pipeline_state_desc(&pso_desc, context.root_signature, 0, NULL, NULL, NULL);
+    pso_desc.StreamOutput.NumEntries = ARRAY_SIZE(so_declaration);
+    pso_desc.StreamOutput.pSODeclaration = so_declaration;
+    pso_desc.StreamOutput.pBufferStrides = strides;
+    pso_desc.StreamOutput.NumStrides = ARRAY_SIZE(strides);
+    pso_desc.StreamOutput.RasterizedStream = D3D12_SO_NO_RASTERIZED_STREAM;
+    hr = ID3D12Device_CreateGraphicsPipelineState(device, &pso_desc,
+            &IID_ID3D12PipelineState, (void **)&context.pipeline_state);
+    if (hr == E_NOTIMPL)
+    {
+        skip("Stream output is not supported.\n");
```

Comment:
Are we ever going to run into this?

Reply:
Copy-paste from the vertex buffer streamout test.

Reply:
Doesn't really matter either way.